packages/vite/src/assets.ts

```diff
@@ -15,7 +15,9 @@ function findPublicAsset(relativePath: string, resolver: Resolver) {
     for (const addon of engine.activeAddons) {
       pkg = packageCache.ownerOfFile(addon.root);
       if (pkg && pkg.meta && pkg.isV2Addon() && pkg.meta['public-assets']) {
-        const asset = Object.entries(pkg.meta['public-assets']).find(([_key, a]) => a === relativePath)?.[0];
+        const asset = Object.entries(pkg.meta['public-assets']).find(
+          ([_key, a]) => join('.', a) === join('.', relativePath)
```

Comment:
i'm not sure i understand this.
when src url is e.g. `'./demo-static-assets/img-first-book/artwork-premier-resto.png'`
the request will still be `localhost:4200/demo-static-assets/img-first-book/artwork-premier-resto.png`
unless the public-assets of an addon somehow have `./` ?

Reply:
ah I found your demo repo @BlueCutOfficial and tested it, thats whats happening!!
![image](https://github.com/embroider-build/embroider/assets/1332320/a07f10b8-e90f-43e7-a73e-661c2d791cc7)

Reply:
> unless the public-assets of an addon somehow have ./

That's the problem, yes, let me re-expose it in details:

I have a classic addon called  `demo-static-assets` that contains a couple of public images. These images are displayed in a `<Gallery />` component provided by the addon. The images are displayed with `img` tags:
```hbs
{{!-- I define rootUrl with the same code as ember-welcome-page --}}
<img src="{{this.rootUrl}}demo-static-assets/.../artwork-gigi-beret.png" alt="..." />
{{!-- I also add an image without the rootUrl as suggested by the Ember CLI docs --}}
<img src="demo-static-assets/.../artwork-cerisier.png" alt="..." />
```

I install this addon in the `vite-app` and invoke the `<Gallery />` component on the application page. When the build starts, `demo-static-assets` is written in the rewritten packages (because it's classic), and the `package.json` contains this, with a `./`:
```
"public-assets": {
      "./public/demo-static-assets/.../artwork-cerisier.png": "./demo-static-assets/.../artwork-cerisier.png",
      "./public/demo-static-assets/.../artwork-gigi-beret.png": "./demo-static-assets/.../artwork-gigi-beret.png"
    }
```

Then when I run or build the `vite-app` with `pnpm start` or `pnpm build` and one of my images is requested, the URL of the request will start with `/demo-static-assets` without the `./` so if we compare this URL to the exact right path in the `package.json` to say "hey, I found asset", my images are never found because we compare `/` to `./`. 

👉 I don't know if the "real" problem is the strict equality or what is generated in the rewritten package, I assume acting on the strict equality and be a little more flexible about this could be better.